lib/matplotlib/colors.py

```diff
@@ -232,9 +232,35 @@ def is_color_like(c):
 
 
 def _has_alpha_channel(c):
-    """Return whether *c* is a color with an alpha channel."""
-    # 4-element sequences are interpreted as r, g, b, a
-    return not isinstance(c, str) and len(c) == 4
+    """
+    Return whether *c* is a color with an alpha channel.
+
+    If *c* is not a valid color specifier, then the result is undefined.
+    """
+    # If c isn't a color, this is undefined. Return False instead of raising.
+    if not is_color_like(c):
+        return False
```

Comment:
```suggestion
    # The following logic is build with the assumtion that c is a valid color spec.
    # For speed and simplicity, we intentionally don't care about other inputs.
    # Anything can happen with them.
```
I believe you can leave this check out. We explicitly state that the result is undefined if `c` is not a valid color specified, so we don't have to expicitly check that case.
I'm a bit picky about this because `is_color_like()` is somewhat expensive, and `_has_alpha_channel()` is quite a low-level function that I want to be fast.

Reply:
Ah sorry that was my suggestion.  Agree that performance is more important.